standalone-metastore/metastore-common/src/main/thrift/hive_metastore.thrift

```diff
@@ -2533,6 +2584,8 @@ service ThriftHiveMetastore extends fb303.FacebookService
   void drop_table_with_environment_context(1:string dbname, 2:string name, 3:bool deleteData,
       4:EnvironmentContext environment_context)
                        throws(1:NoSuchObjectException o1, 2:MetaException o3)
+  void drop_table_req(1:DropTableRequest dropTableReq)
+        throws(1:NoSuchObjectException o1, 2:MetaException o3)
   void truncate_table(1:string dbName, 2:string tableName, 3:list<string> partNames)
```

Comment:
should we convert this to a request based as well?

Reply:
Request-based API for this already exists. See L2591.